### PR TITLE
Integrate HW-MGMT 7.0040.1002 Changes

### PR DESCRIPTION
**Pull request by keboliu**
#### Why I did it

Integrate HW-MGMT 7.0040.1002 Changes

 ## Patch List
* 0091-platform-mellanox-mlxreg-dpu-Add-initial-support-for.patch :
* 0092-platform-mellanox-Introduce-support-of-Nvidia-smart-.patch :
* 0094-i2c-asf-Introduce-MCTP-support-over-ASF-controller.patch :

#### How I did it
Run make integrate-mlnx-hw-mgmt

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@saiarcot895 could you please help to review and merge?
please also refer to the backport PR

File: patch/kconfig-inclusions [hunk 1]
```diff
@@ -113,10 +119,15 @@ CONFIG_SENSORS_LM25066=m
...
 CONFIG_THERMAL_OF=y
+CONFIG_MCTP_FLOWS=y
+CONFIG_MCTP_TRANSPORT_I2C=m
+CONFIG_I2C_SLAVE_EEPROM=m
+CONFIG_I2C_ASF=m
 CONFIG_SPI_AMD=m
 CONFIG_PINCTRL_AMD=y
 CONFIG_EDAC_AMD64=m
 CONFIG_AMD_XGBE_DCB=y
+CONFIG_USB_NET_CDCETHER=m
```
Comment (saiarcot895) [inline patch/kconfig-inclusions:130]: Same comment as https://github.com/sonic-net/sonic-linux-kernel/pull/396#discussion_r1625191520

File: patch/kconfig-inclusions [hunk 2]
```diff
@@ -35,12 +35,16 @@ CONFIG_SENSORS_MAX31790=m
 CONFIG_EEPROM_OPTOE=m
 ###-> mellanox_amd64-start
 CONFIG_X86_AMD_PLATFORM_DEVICE=y
+CONFIG_MCTP=y
+CONFIG_HOTPLUG_PCI_PCIE=n
```
[the remark this answers is not in the document]
Comment (prgeor) [reply patch/kconfig-inclusions:39]: I think PCIe hotplug is used by T2 chassis for line card insertion/removal. @abdosi fyi

[the remark this answers is not in the document]
Comment (saiarcot895) [reply patch/kconfig-inclusions:39]: One suggestion: I don't know if this is even possible with PCIe, but is it possible to disable hotplug functionality for the bridge that is connected to the ASIC?

**Pull request closed without merging**